spring-data-mongodb/src/test/java/org/springframework/data/mongodb/core/aggregation/AggregationTests.java

```diff
@@ -1018,6 +1021,34 @@ public void shouldRetrieveDateTimeFragementsCorrectly() throws Exception {
 		assertThat(dbo.get("dayOfYearPlus1DayManually"), is((Object) dateTime.plusDays(1).getDayOfYear()));
 	}
 
+	/**
+	 * @see DATAMONGO-1127
+	 */
+	@Test
+	public void shouldSupportGeoNearQueriesForAggregation() {
+
+		mongoTemplate.dropCollection(Venue.class);
```

Comment:
We should handle these cleanup operations in JUnit callback methods (the same for the drop of the collection at the very end) to make sure they get applied even if exceptions occur.